lib/client/bespoke-client.ts

```diff
@@ -46,12 +44,28 @@ export class BespokeClient {
                 self.messageReceived(data, messageID);
             }
         );
+    }
+    public connect(onConnect?: (error?: any) => void): void {
+        const self = this;
+
+        if (onConnect !== undefined && onConnect !== null) {
+            this.onConnect = onConnect;
+        }
+
+        this.attemptConnection();
 
         // If the socket gets closed, probably server-side issue
         // We do not do anything in this case other than
         this.socketHandler.onCloseCallback = function() {
```

Comment:
I believe this block needs to be part of the onCloseCallback - that creates a new socketHandler and so this must be set on that socketHandler.